asserts/database.go

```diff
@@ -89,6 +89,27 @@ var (
 	ErrNotFound = errors.New("assertion not found")
 )
 
+// RevisionError indicates a revision improperly used for an operation.
+type RevisionError struct {
+	Used, Current int
+}
+
+func (e *RevisionError) Error() string {
+	if e.Used < 0 {
+		return fmt.Sprintf("used assertion revision %d is unknown", e.Used)
+	}
+	if e.Current < 0 {
```

Comment:
But then {Used: 2, Current: -1} would produce a message of, "assertion revision 2 is unknown", that doesn't make sense.

We could drop the first word - used or current - from the message but with the invalid revision's value?